apm-sniffer/apm-sdk-plugin/xxl-job-2.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/xxljob/SimpleJobHandlerMethodInterceptor.java

```diff
@@ -38,13 +40,20 @@ public class SimpleJobHandlerMethodInterceptor implements InstanceMethodsAroundI
 
     @Override
     public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes, MethodInterceptResult result) throws Throwable {
-        String jobParam = (String) allArguments[0];
         String operationName = ComponentsDefine.XXL_JOB.getName() + "/SimpleJob/" + method.getDeclaringClass().getName();
 
         AbstractSpan span = ContextManager.createLocalSpan(operationName);
         span.setComponent(ComponentsDefine.XXL_JOB);
         Tags.LOGIC_ENDPOINT.set(span, Tags.VAL_LOCAL_SPAN_AS_LOGIC_ENDPOINT);
-        span.tag(JOB_PARAM, jobParam);
+        if (allArguments.length == 1) {
+            String jobParam = (String) allArguments[0];
+            span.tag(JOB_PARAM, jobParam);
+        } else if (allArguments.length == 0) {
+            Class<?> xxlJobHelper = Class.forName(XXL_JOB_HELPER);
+            Method getJobParamMethod = xxlJobHelper.getMethod(XXL_JOB_HELPER_GET_PARAM_METHOD);
```

Comment:
Is this for 2.3.0 only? If so, you should be able to upgrade compiling version to 2.3.0, and this uses method directly. When `length == 1`, this would not be executed, and no exception would occur.